src/patternfly/components/Tabs/tabs.scss

```diff
@@ -112,167 +126,305 @@
+  .pf-c-tabs__item-text {
+    margin-right: var(--pf-c-tabs__button--child--MarginRight);
 
-  &:first-of-type {
-    left: 0;
+    &:last-child {
+      --pf-c-tabs__button--child--MarginRight: 0;
+    }
   }
 
-  &:last-of-type {
-    right: 0;
+  .pf-c-tabs.pf-m-roomy & {
+    --pf-c-tabs__button--PaddingRight: var(--pf-c-tabs--m-roomy--c-tabs__button--PaddingRight);
+    --pf-c-tabs__button--PaddingLeft: var(--pf-c-tabs--m-roomy--c-tabs__button--PaddingLeft);
   }
 }
 
 // Scroll buttons
 .pf-c-tabs__scroll-button {
-  position: absolute;
-  top: 0;
-  bottom: 0;
-  z-index: var(--pf-c-tabs__scroll-button--ZIndex);
+  flex: none;
   width: var(--pf-c-tabs__scroll-button--Width);
   line-height: 1;
-  background-color: var(--pf-c-tabs__item--BackgroundColor);
-  border: 0;
-
-  // Align right scroll button
-  &:nth-of-type(2) {
-    right: 0;
+  color: var(--pf-c-tabs__scroll-button--Color);
+  visibility: hidden;
+  background-color: var(--pf-c-tabs__scroll-button--BackgroundColor);
+  outline-offset: var(--pf-c-tabs__scroll-button--OutlineOffset);
+  opacity: 0;
+  transition: var(--pf-c-tabs__scroll-button--Transition);
+
+  &:hover,
+  &:active,
+  &:focus {
+    --pf-c-tabs__scroll-button--Color: var(--pf-c-tabs__scroll-button--hover--Color);
   }
 
-  // Scroll button borders
-  &:not(.pf-m-secondary) {
+  &:nth-of-type(1) {
+    margin-right: calc(var(--pf-c-tabs__scroll-button--Width) * -1);
+    transform: translateX(-100%);
+
     &::before {
-      position: absolute;
-      top: 0;
-      right: 0;
-      bottom: 0;
-      left: 0;
-      content: "";
-      border: var(--pf-c-tabs__item--BorderWidth) solid var(--pf-c-tabs__item--BorderColor);
+      border-right-color: var(--pf-c-tabs__scroll-button--before--BorderColor);
+      border-right-width: var(--pf-c-tabs__scroll-button--before--BorderWidth);
     }
+  }
 
-    &:hover::after {
-      position: absolute;
-      top: 0;
-      right: 0;
-      bottom: 0;
-      left: 0;
-      content: "";
-      border-top: var(--pf-c-tabs__item--m-current--BorderTopWidth) solid var(--pf-c-tabs__item--hover--Color);
+  &:nth-of-type(2) {
+    margin-left: calc(var(--pf-c-tabs__scroll-button--Width) * -1);
+    transform: translateX(100%);
+
+    &::before {
+      border-left-color: var(--pf-c-tabs__scroll-button--before--BorderColor);
+      border-left-width: var(--pf-c-tabs__scroll-button--before--BorderWidth);
     }
   }
 
+  &:disabled,
+  &:disabled:hover {
+    --pf-c-tabs__scroll-button--Color: var(--pf-c-tabs__scroll-button--disabled--Color);
+
+    pointer-events: none;
+  }
+
+  &::after {
+    content: none;
+  }
+}
 
-  &.pf-m-secondary {
-    &.pf-m-hover,
-    &:hover,
-    &.pf-m-active,
-    &:active,
-    &.pf-m-focus,
-    &:focus {
-      color: var(--pf-c-tabs__scroll-button--m-secondary--hover--Color);
+// Scrollable, scroll buttons enabled
+.pf-c-tabs.pf-m-scrollable {
+  .pf-c-tabs__scroll-button {
+    visibility: visible;
+    opacity: 1;
+
+    &::before {
```

Comment:
should this move up to the base `.pf-c-tabs__scroll-button` styles so we're only changing what's needed with `.pf-m-scrollable`?